packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2501,5 +2506,121 @@ describe("Runtime", () => {
 			);
 			assert.equal(mockLogger.events.length, 0, "Expected no more events logged");
 		});
+
+		describe("Signals", () => {
+			let containerRuntime: ContainerRuntime;
+			let logger: MockLogger;
+			beforeEach(async () => {
+				logger = new MockLogger();
+				containerRuntime = await ContainerRuntime.loadRuntime({
+					context: getMockContext({}, logger) as IContainerContext,
+					registryEntries: [],
+					existing: false,
+					requestHandler: undefined,
+					runtimeOptions: {
+						enableGroupedBatching: false,
+					},
+					provideEntryPoint: mockProvideEntryPoint,
+				});
+				clock.restore();
+			});
+
+			function sendSignals(numberOfSignals: number = 1) {
+				for (let i = 0; i < numberOfSignals; i++) {
+					containerRuntime.submitSignal("TestSignalType", "TestSignalContent");
+				}
+			}
+
+			function processSignals(signalSequenceNumber: number, numberOfSignals: number = 1) {
+				for (let i = 0; i < numberOfSignals; i++) {
+					containerRuntime.processSignal(
+						{
+							clientId: containerRuntime.clientId as string,
+							content: {
+								clientSignalSequenceNumber: signalSequenceNumber + i,
+								contents: {
+									type: "TestSignalType",
+									content: "TestSignalContent",
+								},
```

Comment:
I think this is getting closer and good to be unit test level. But this processSignal call backs are completely disjointed from whatever the containerRuntime chooses to send. It would be okay for ContainerRuntime to start its `clientSignalSequenceNumber` at any value. Then the tests would be broken.
Where do the `submitSignal` calls go? Can we watch that can configure how it feeds back to ContainerRuntime?

Reply:
I changed it so now the container runtime consumes the actual signals that are submitted, so that `clientSignalSequenceNumber` is not hard coded to be a specific number. The submitted signals are stored in a queue and can be either processed or 'dropped'. We can then maintain a list of 'dropped' signals to make them reappear out of order also.